lib/jormungandr/src/Cardano/Wallet/Jormungandr/Transaction.hs

```diff
@@ -1,19 +1,67 @@
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
 
 module Cardano.Wallet.Jormungandr.Transaction
     ( newTransactionLayer
     ) where
 
 import Prelude
 
+import Cardano.Wallet.Jormungandr.Compatibility
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( Depth (AddressK), Key, Passphrase (..), XPrv, getKey )
+import Cardano.Wallet.Primitive.Types
+    ( Hash (..), Tx (..), TxId (..), TxOut (..), TxWitness (..) )
 import Cardano.Wallet.Transaction
-    ( TransactionLayer (..) )
+    ( ErrMkStdTx (..), TransactionLayer (..) )
+import Control.Monad
+    ( forM )
+import Data.ByteString
+    ( ByteString )
 
+import qualified Cardano.Crypto.Wallet as CC
 
 -- | Construct a 'TransactionLayer' compatible with Shelley and 'Jörmungandr'
-newTransactionLayer :: TransactionLayer t
-newTransactionLayer = TransactionLayer
-    { mkStdTx = error "TODO: See http-bridge as starting point"
+newTransactionLayer
+    :: forall n . Hash "Block0Hash"
```

Comment:
@KtorZ 

> I am not sure about this type ... And, does this really have to be the hash from the first block header?

To my knowledge — yes

Reply:
https://github.com/input-output-hk/rust-cardano/blob/e5d974f7bedeb00c9c9d688ac66094a34bf8f40d/chain-impl-mockchain/src/transaction/witness.rs#L60-L62